### DISCUSSION TITLE
Portable Sink Plugin ERROR

### DISCUSSION THREAD

**Discussion by erniu831** [lf-edge/ekuiper#2411] [Q&A]
I try to create a sink plugin
`{
  "about": {
    "trial": false,
    "author": {
      "name": "J",
      "email": "xxx",
      "company": "xxx"
    },
    "description": {
      "en_US": "This a plugin for k-v state."
    }
  },
  "libs": [],
  "properties": [
      {
        "name": "key",
        "default": "default",
        "optional": true,
        "control": "text",
        "type": "string",
        "hint": {
          "en_US": "the interval which you want to send data to mongodb."
        },
        "label": {
          "en_US": "state key",
          "zh_CN": "存储键"
        }
      }
  ]
}`

but when i press this "test connection" button on ekuiper-manager, got ERROR is:
`receive error: invalid arg &{stateGo 0x40000a3080 sink map[bufferLength:1024 enableCache:false format:json key:default omitIfEmpty:false runAsync:false sendSingle:true]}, ruleId and opId are required`

this is my code snippet
```
type stateSink struct {
	sync.Mutex
	cancel context.CancelFunc
	key    string
}

func (n *stateSink) Open(ctx api.StreamContext) error {
	// Do Nothing
	return nil
}

func (n *stateSink) Configure(props map[string]interface{}) error {
	if i, ok := props["key"]; ok {
		if i, ok := i.(string); ok {
			n.key = i
		} else {
			return errors.New("key Must be string")
		}
	}
	return nil
}

func (n *stateSink) Collect(ctx api.StreamContext, data interface{}) error {
	// DO SOMETHING
	return nil
}

func (n *stateSink) Close(ctx api.StreamContext) error {
	if n.cancel != nil {
		n.cancel()
	}
	return nil
}
```

I am puzzled and hope to get some answers. 
Eager for reply, Thanks.

**Comment by ngjaying** [comment]
It seems portable plugin is not adapted to the "test connection" interface so the feature is not working.
